File: base-v3/src/main/java/com/adyen/checkout/base/Configuration.java
/*
 * Copyright (c) 2019 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by caiof on 7/3/2019.
 */

package com.adyen.checkout.base;

import android.support.annotation.NonNull;

import com.adyen.checkout.core.api.Environment;

import java.util.Locale;

/**
 * Configuration class with extra information the merchant can or should provide to change the {@link PaymentComponent} behavior.
 */
public interface Configuration {

    @NonNull
    Locale getShopperLocale();

    @NonNull
    Environment getEnvironment();
}

File: components-core/src/main/java/com/adyen/checkout/components/base/Configuration.kt
package com.adyen.checkout.components.base

import android.os.Parcelable
import com.adyen.checkout.components.model.payments.Amount
import com.adyen.checkout.core.api.Environment
import java.util.Locale

interface Configuration : Parcelable {

    val shopperLocale: Locale
    val environment: Environment
    val clientKey: String
    val isAnalyticsEnabled: Boolean?
    val amount: Amount
}
